src/vs/workbench/contrib/chat/common/chatEntitlementService.ts

```diff
@@ -1004,9 +1025,10 @@ export class ChatEntitlementContext extends Disposable {
 	update(context: { installed: boolean; disabled: boolean; untrusted: boolean }): Promise<void>;
 	update(context: { hidden: boolean }): Promise<void>;
 	update(context: { later: boolean }): Promise<void>;
-	update(context: { entitlement: ChatEntitlement }): Promise<void>;
-	update(context: { installed?: boolean; disabled?: boolean; untrusted?: boolean; hidden?: boolean; later?: boolean; entitlement?: ChatEntitlement }): Promise<void> {
+	update(context: { entitlement: ChatEntitlement; isInternal?: boolean }): Promise<void>;
+	update(context: { installed?: boolean; disabled?: boolean; untrusted?: boolean; hidden?: boolean; later?: boolean; entitlement?: ChatEntitlement; isInternal?: boolean }): Promise<void> {
 		this.logService.trace(`[chat entitlement context] update(): ${JSON.stringify(context)}`);
+		this.state.isInternal = !!context.isInternal;
```

Comment:
@lramos15 this is not good, there are calls into this method without the `isInternal` flag and thus they would overwrite this flag by accident. Please take a look at my follow up change in https://github.com/microsoft/vscode/pull/261596